workflow-core/src/commonMain/kotlin/com/squareup/workflow1/WorkflowAction.kt

```diff
@@ -22,15 +22,25 @@ public abstract class WorkflowAction<in PropsT, StateT, out OutputT> {
     public val props: @UnsafeVariance PropsT,
     public var state: StateT
   ) {
-    internal var maybeOutput: WorkflowOutput<@UnsafeVariance OutputT>? = null
+    internal val startingState = state
+    internal var forcedRerender: Boolean = false
+    internal var outputOrNull: WorkflowOutput<@UnsafeVariance OutputT>? = null
       private set
 
     /**
      * Sets the value the workflow will emit as output when this action is applied.
      * If this method is not called, there will be no output.
      */
     public fun setOutput(output: @UnsafeVariance OutputT) {
-      this.maybeOutput = WorkflowOutput(output)
+      this.outputOrNull = WorkflowOutput(output)
+    }
+
+    /**
+     * The action will cause the Workflow to re-render regardless of whether or not state has
+     * changed, for all [RuntimeConfig]s.
+     */
+    public fun forceRerender() {
+      forcedRerender = true
```

Comment:
Surprised this exists, and surprised that it's public. Hard to imagine feature devs putting this to anything but bad uses.

Reply:
I originally did not have this. I've now added this as in internal testing we have a number of scenarios where we drive 'other' (non-Workflow) UI via side effects that occur in actions, but then that require a re-render() to pick up the changes in that other UI. So either we add some kind of increasing counter to the state for those cases or we add a hook to force re-render() without state change like this. I wanted to just test with this first. Sorry I'm realizing now that this PR shoudl still be in draft. I'm not sure why it wasn't. Probably I was thinking at some point it was more ready than it is.